pyhive/hive.py

```diff
@@ -79,20 +82,29 @@ def __init__(self, host, port=10000, username=None, database='default', auth='NO
         if auth == 'NOSASL':
             # NOSASL corresponds to hive.server2.authentication=NOSASL in hive-site.xml
             self._transport = thrift.transport.TTransport.TBufferedTransport(socket)
-        elif auth == 'NONE':
+        elif auth.upper() in ['LDAP', 'PLAIN', 'GSSAPI', 'NONE']:
```

Comment:
PLAIN and GSSAPI aren't valid values for hive.server2.authentication (see docstring). Can we just have one supported way? (no auto-remapping and no auto-casing)